base/logging_test.go

```diff
@@ -242,3 +244,140 @@ func TestLogColor(t *testing.T) {
 	assert.Equal(t, "Format", color("Format", LevelTrace))
 	assert.Equal(t, "Format", color("Format", LevelNone))
 }
+
+func TestSGLevel(t *testing.T) {
+	assert.Equal(t, 1, DebugLevel.sgLevel())
+	assert.Equal(t, 1, InfoLevel.sgLevel())
+	assert.Equal(t, 2, WarnLevel.sgLevel())
+	assert.Equal(t, 2, ErrorLevel.sgLevel())
+	assert.Equal(t, 3, PanicLevel.sgLevel())
+	assert.Equal(t, 3, FatalLevel.sgLevel())
+	assert.Equal(t, 4, Level(5).sgLevel())
+}
+
+func TestMarshalTextError(t *testing.T) {
+	var level *Level
+	bytes, err := level.MarshalText()
+	assert.Nil(t, bytes, "bytes should be nil")
+	assert.Error(t, err, "Can't marshal a nil *Level to text")
+	assert.Equal(t, err.Error(), "can't marshal a nil *Level to text")
+}
+
+func TestGetCallersNameRecoverInfoImpossible(t *testing.T) {
+	callerName := GetCallersName(3, true)
+	assert.Equal(t, "???", callerName)
+	callerName = GetCallersName(3, false)
+	assert.Equal(t, "???", callerName)
+}
+
+func TestLastComponent(t *testing.T) {
+	path := lastComponent("/var/log/sync_gateway/sglogfile.log")
+	assert.Equal(t, "sglogfile.log", path)
+	path = lastComponent("\\var\\log\\sync_gateway\\sglogfile.log")
+	assert.Equal(t, "sglogfile.log", path)
+	path = lastComponent("sglogfile.log")
+	assert.Equal(t, "sglogfile.log", path)
+	path = lastComponent("/sglogfile.log")
+	assert.Equal(t, "sglogfile.log", path)
+	path = lastComponent("\\sglogfile.log")
+	assert.Equal(t, "sglogfile.log", path)
+}
+
+func TestLogSyncGatewayVersion(t *testing.T) {
+	out := CaptureOutput(t, LogSyncGatewayVersion)
+	assert.Contains(t, out, LongVersionString)
+	logger := &FileLogger{Enabled: true, logger: &log.Logger{}}
+
+	logger.level = LevelNone
+	assert.True(t, logger.shouldLog(LevelNone))
+	assert.False(t, logger.shouldLog(LevelWarn))
+	assert.False(t, logger.shouldLog(LevelError))
+	assert.False(t, logger.shouldLog(LevelInfo))
+	assert.False(t, logger.shouldLog(LevelDebug))
+	assert.False(t, logger.shouldLog(LevelTrace))
+	out = CaptureOutput(t, LogSyncGatewayVersion)
+	assert.Contains(t, out, LongVersionString)
+
+	logger.level = LevelWarn
+	assert.True(t, logger.shouldLog(LevelNone))
+	assert.True(t, logger.shouldLog(LevelWarn))
+	assert.True(t, logger.shouldLog(LevelError))
+	assert.False(t, logger.shouldLog(LevelInfo))
+	assert.False(t, logger.shouldLog(LevelDebug))
+	assert.False(t, logger.shouldLog(LevelTrace))
+	out = CaptureOutput(t, LogSyncGatewayVersion)
+	assert.Contains(t, out, LongVersionString)
```

Comment:
Given all of the assertions for each log level are the same, I'm wondering if we can write the test like this:

```golang
// Tests that LogSyncGatewayVersion writes output to the console for each log level.
func TestLogSyncGatewayVersion(t *testing.T) {
	for i := LevelNone; i < levelCount; i++ {
		t.Run(i.String(), func(t *testing.T) {
			consoleLogger.LogLevel.Set(i)
			out := CaptureOutput(LogSyncGatewayVersion)
			assert.Contains(t, out, LongVersionString)
		})
	}
}
```

Reply:
Yeah, this would be better to the old approach.

Reply:
Changes are committed to the latest version.